libdrgn/orc_info.c

```diff
@@ -152,11 +152,15 @@ remove_fdes_from_orc(struct drgn_module *module, unsigned int *indices,
 	unsigned int num_entries = *num_entriesp;
 	unsigned int new_num_entries = 0;
 
-	uint64_t start_pc = drgn_raw_orc_pc(module, 0);
+	// ORC can be built-in or from the debug file. Because of that, we
+	// always store the biased/actual address at orc.pc_base. Since we are
+	// comparing to the unbiased addresses in the debug_frame FDEs, we need
+	// to subtract the bias from the ORC PC.
+	uint64_t start_pc = drgn_raw_orc_pc(module, 0) - module->debug_file_bias;
```

Comment:
It's somewhat confusing that `drgn_module_orc_info::preferred` contains unbiased addresses and `drgn_orc_pc()`/`drgn_raw_orc_pc()` is biased. `drgn_module_should_prefer_orc_cfi()` expects unbiased addresses, so I think it's correct as you've written it, just slightly awkward. Do you think we should change it to all be biased? I'm on the fence.

I'm going to add a test case that I should have added in 98995b617d4479daf4846a18b3ce97cbaa554628...